app/src/validators/permission.js

```diff
@@ -0,0 +1,60 @@
+const { validate, Joi } = require('express-validation');
+const { uuidv4, uuidv4MultiModel, permCodeMultiModel } = require('./common');
+const { Permissions } = require('../components/constants');
+
+const schema = {
+  searchPermissions: {
+    query: Joi.object({
+      userId: uuidv4MultiModel,
+      objId: uuidv4MultiModel,
+      permCode: permCodeMultiModel
+    }).min(1)
+  },
+
+  listPermissions: {
+    query: Joi.object({
+      userId: uuidv4MultiModel,
+      objId: uuidv4MultiModel,
+      permCode: permCodeMultiModel
+    })
+  },
+
+  addPermissions: {
+    params: Joi.object({
+      objId: uuidv4
+    }),
+    body: Joi.array().items(
+      Joi.object().keys({
+        userId: uuidv4.required(),
+        permCode: Joi.string().max(255).required().valid(
+          Permissions.CREATE,
+          Permissions.READ,
+          Permissions.UPDATE,
+          Permissions.DELETE,
+          Permissions.MANAGE
+        ),
+      })
+    ).required(),
+  },
+
+  removePermissions: {
+    params: Joi.object({
+      objId: uuidv4
+    }),
+    query: Joi.object({
+      userId: uuidv4MultiModel,
```

Comment:
a userId and permCode is also required according to the code in the removePermissions() controller.